tests/unistd/truncate.c

```diff
@@ -79,15 +79,24 @@ int main() {
 
   printf("truncate(2): %d\n", truncate("towrite", 2));
   printf("errno: %s\n", strerror(errno));
-  stat("towrite", &s);
+  fstat(f, &s);
+  printf("st_size: %lld\n", s.st_size);
+  memset(&s, 0, sizeof s);
+  errno = 0;
+  printf("\n");
+
+  printf("open(O_TRUNC)\n");
+  open("towrite", O_RDONLY | O_TRUNC);
+  printf("errno: %s\n", strerror(errno));
+  fstat(f, &s);
   printf("st_size: %lld\n", s.st_size);
   memset(&s, 0, sizeof s);
   errno = 0;
   printf("\n");
 
   printf("truncate(readonly, 2): %d\n", truncate("toread", 2));
   printf("errno: %s\n", strerror(errno));
-  stat("toread", &s);
+  fstat(f2, &s);
```

Comment:
Why not leave one check for `stat` for more coverage?

Reply:
sgtm, but please confirm `stat` is actually covered. I don't remember offhand.

Reply:
Confirmed! We even have a whole test directory tests/stat.